.cbmc-batch/jobs/aws_ring_buffer_buf_belongs_to_pool/aws_ring_buffer_buf_belongs_to_pool_harness.c

```diff
@@ -26,14 +26,26 @@ void aws_ring_buffer_buf_belongs_to_pool_harness() {
 
     /* assumptions */
     ensure_ring_buffer_has_allocated_members(&ring_buf, ring_buf_size);
+    bool is_member = nondet_bool(); /* nondet assignment required to force true/false */
```

Comment:
I don't understand this comment.  Shouldn't it nondet to bool already?

Reply:
Yes, I thought so too. But, I get a counterexample without the nondet assignment. In the following, CBMC will assign a truthy value (like 2) to x:

```
#include <stdbool.h>
int main() {
   bool x;
   assert(x == true || x == false); //< fails
}
```

Reply:
Since C doesn't have single-bit types, `bool` (which typically is defined to be `_Bool`) is at least eight bits (on common platforms), and CBMC will happily pick a value that is neither `0` (false) nor `1` (true). `nondet_bool` in CBMC is special (see https://github.com/diffblue/cbmc/blob/306697d69605bb9b3c7f2e60a9a71cc9dff65419/src/goto-programs/builtin_functions.cpp#L875) in that it forces the return value to be 0 or 1.